qa/tasks/cephfs/test_volumes.py

```diff
@@ -136,6 +136,167 @@ def test_subvolume_create_and_rm(self):
+
+    def test_subvolume_resize_fail_invalid_size(self):
+        # tests the 'fs subvolume resize' command for an invalid size
+
+        # create subvolume
+        subvolname = self._generate_random_subvolume_name()
+        self._fs_cmd("subvolume", "create", self.volname, subvolname, "--size",
+                     str(self.DEFAULT_FILE_SIZE*1024*1024))
+
+        # make sure it exists
+        subvolpath = self._get_subvolume_path(self.volname, subvolname)
+        self.assertNotEqual(subvolpath, None)
+
+        # try to resize the subvolume with an invalid size -10
+        nsize = -10
+        try:
+            self._fs_cmd("subvolume", "resize", self.volname, subvolname, str(nsize))
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.EINVAL:
+                raise
+        else:
+            raise RuntimeError("expected the 'fs subvolume resize' command to fail")
+
+    def test_subvolume_resize_fail_zero_size(self):
+        # tests the 'fs subvolume resize' command for size 0
+
+        # create subvolume
+        subvolname = self._generate_random_subvolume_name()
+        self._fs_cmd("subvolume", "create", self.volname, subvolname, "--size",
+                     str(self.DEFAULT_FILE_SIZE*1024*1024))
+
+        # make sure it exists
+        subvolpath = self._get_subvolume_path(self.volname, subvolname)
+        self.assertNotEqual(subvolpath, None)
+
+        # try to resize the subvolume with size 0
+        nsize = 0
+        try:
+            self._fs_cmd("subvolume", "resize", self.volname, subvolname, str(nsize))
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.EINVAL:
+                raise
+        else:
+            raise RuntimeError("expected the 'fs subvolume resize' command to fail")
+
+    def test_subvolume_resize_fail_quota_smaller_than_used_size(self):
+        # tests the 'fs subvolume resize' command for setting new size smaller than
+        # the current used size
+
+        osize = self.DEFAULT_FILE_SIZE*1024*1024*2
+        # create subvolume
+        subvolname = self._generate_random_subvolume_name()
+        self._fs_cmd("subvolume", "create", self.volname, subvolname, "--size", str(osize))
+
+        # make sure it exists
+        subvolpath = self._get_subvolume_path(self.volname, subvolname)
+        self.assertNotEqual(subvolpath, None)
+
+        #create one file of 1MB
+        file_size=self.DEFAULT_FILE_SIZE
+        number_of_files=1
+        log.debug("filling subvolume {0} with {1} file of size {2}MB".format(subvolname,
+                                                                             number_of_files,
+                                                                             file_size))
+        filename = "{0}.{1}".format(TestVolumes.TEST_FILE_NAME_PREFIX, 0)
+        self.mount_a.write_n_mb(os.path.join(subvolpath, filename), file_size)
+
+        usedsize = int(self.mount_a.getfattr(subvolpath, "ceph.dir.rbytes"))
```

Comment:
Verify `usedsize` is approximately `self.DEFAULT_FILE_SIZE`.

Reply:
@batrick They are integers. So I think you mean check the equality?

Reply:
The rstat may not be exact but close to the correct value.